sdk/provisioning/Generator/src/Model/Specification.cs

```diff
@@ -35,7 +35,9 @@ public abstract partial class Specification : ModelBase
     public Dictionary<string, ModelBase> ModelNameMapping { get; } = [];
     public Dictionary<Type, ModelBase> ModelArmTypeMapping { get; } = [];
 
-    public Specification(string name, Type armEntryPoint)
+    internal bool IgnorePropertiesWithoutPath { get; }
+
+    public Specification(string name, Type armEntryPoint, bool ignorePropertiesWithoutPath = false)
```

Comment:
I added this new option which should benefit us in the future.
Now only network is using the value of true, those old RPs keep their previous behavior of false.

This flag will make the generator no longer generates the properties without a WirePath attribute.
Previously all the properties on models would be generated, including those hidden or for convenience.